app/models/time_entry.rb

```diff
@@ -53,20 +53,21 @@ class TimeEntry < ActiveRecord::Base
   validate :validate_project_is_set
   validate :validate_consistency_of_work_package_id
 
-  scope :visible, ->(*args) {
+
+  scope :on_work_packages, ->(work_packages) { where(work_package_id: work_packages) }
+
+  after_initialize :set_default_activity
+  before_validation :set_default_project
+
+  def self.visible(*args)
```

Comment:
Avoiding a scope here makes it clear what the `CostScope` in [this PR](https://github.com/finnlabs/openproject-costs/pull/299) actually overrides. It would have worked with `scope :visible` as well since that's only some sugar around `define_method`, however.